### ISSUE TITLE
builder should support "self-signed" certs provided by the Corporate Firewall

### ISSUE THREAD

**Issue by bdangit** [habitat-sh/on-prem-builder#114]
My OAuth endpoint is behind our Corporate Firewall and we use our own Corporate Cert.  Every time, I try to sign-in, I get the following error from `builder-api`:
```
Aug 29 23:13:32 localhost.localdomain hab[664]: builder-api.default(O):  WARN 2018-08-29T23:13:32Z: habitat_builder_api::server::handlers: Oauth client error, HttpClient(Error { kind: Io(Custom { kind: Other, error: Ssl(ErrorStack([Error { code: 336134278, library: "SSL routines", function: "ssl3_get_server_certificate", reason: "certificate verify failed", file: "s3_clnt.c", line: 1269 }])) }), url: Some("https://<redacted>/") })
```

I tried looking through the code to see how I can make sure any http request uses our corporate cert, but to no avail.

Without this feature, I am unable to start up my own on-premise builder.

**Comment by bdangit** [comment]
spent a bit more time on this.  after messing around with `SSL_CERT_FILE` with keeping it default of `$(hab pkg path core/cacert)/...` and messing with the run hook of the `builder-api`, out of the whim I decided just to start `builder-api` without hab-sup.  

```bash
[root]# HOME="/hab/svc/builder-api/data" RUST_LOG="debug,tokio_core=error,tokio_reactor=error,zmq=error,hyper=error" RUST_BACKTRACE=1 exec /hab/pkgs/core/busybox-static/128.1/20180608102729/bin/chpst -U "hab:hab" -u "hab:hab" /hab/pkgs/habitat/builder-api/7987/20181203194923/bin/bldr-api start -c "/hab/svc/builder-api/config/config.toml" 2>&1
```

I was then able to oauth with on-premise gitlab.  I checked the process environment variables:

```
[root]# cat /proc/104203/environ
RUST_LOG=debug...HOME=/hab/svc/builder-api...RUST_BACKTRACE=1...
```

No mentions of any SSL_CERT_FILE.  Even my terminal environment does not have SSL envvars.

I'm now very perplexed how this is even working.  I'm guessing it's the because of the `builder-api-proxy` ??

**Comment by bdangit** [comment]
As a further test, I placed a `unset SSL_CERT_FILE` before spawning `bldr-api` in the run hook.  This was to negate the SSL_CERT_FILE envvar being set in the `hab-sup.service` systemd unit.

I was successful to oauth.  I even tried another browser with no history or cookie of logging into both bldr and gitlab.  

Again, I am extremely perplexed as to why this is working the way it is.  The debug logs are showing:

```
builder-api.default(O): [... DEBUG oauth_client::gitlab] GitLab response body: ...
builder-api.default(O): [... DEBUG reqest::async_impl::response] Response: '200 OK' for https://<onprem gitlab>/oauth/userinfo
builder-api.default(O): [... DEBUG oauth_client::gitlab] GitLab response body ...
...
```

A few other points about my environment:
I'm running all the on-prem scripts under a Centos with systemd docker container.  I also installed corporate ssl cert into the ca-certificates so that `curl https://google.com`, etc would work as normal.
